### PR TITLE
Refactor Terraform scripts to support other providers

### PR DESCRIPTION
**Pull request by electrocucaracha**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
The current Terraform implementation was not using all the default variables defined as well as it was missing the formatting. This change provides a different layout to allow the addition of more providers in the future. It also fixes the periodic prow job.  

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by electrocucaracha** [comment]
@radoslawc it seems like prow is not detecting the changes on e2e folder? I'm wondering if we broke something

File: e2e/terraform/modules/gcp/main.tf [hunk 1]
```diff
@@ -7,8 +21,8 @@ resource "random_string" "vm-name" {
 }
 
 locals {
-  vm-name = "e2e-vm-${random_string.vm-name.result}"
-  nephioyaml = "/home/${var.home_user}/nephio.yaml"
+  vm-name    = "e2e-vm-${random_string.vm-name.result}"
+  nephioyaml = "/home/${var.ansible_user}/nephio.yaml"
```
Comment (vjayaramrh) [inline e2e/terraform/modules/gcp/main.tf:25]: are you assuming that the ansible_user and the home_user would always be the same?If yes, makes sense to combine

**Review by vjayaramrh — APPROVED** [review]
/lgtm

File: e2e/terraform/modules/gcp/main.tf (hunk 1 above)
Comment (electrocucaracha) [reply e2e/terraform/modules/gcp/main.tf:25]: That's right, there is no sense to have them separately

**Pull request merged**